### PR TITLE
fix: miscl fixes

### PR DESCRIPTION
**Pull request by Manason**
- Spawning vehicles
- Preventing non EMS from interacting with EMS job functions
- proper lib callbacks
- fix a few qbx_medical export renames

### CODE REVIEW

**Review by KostaZx — APPROVED** [review]

File: server/hospital.lua [hunk 1]
```diff
@@ -66,13 +66,15 @@ local function wipeInventory(src)
 end
 
 lib.callback.register('qbx_ambulancejob:server:spawnVehicle', function(source, vehicleName, vehicleCoords)
-	local netId, veh = qbx.spawnVehicle({ spawnSource = vehicleCoords or source, model = vehicleName, warp = source })
+	local src = source
```
Comment (FjamZoo) [inline server/hospital.lua]: Why did you localize source here when its inside the callback?

File: client/job.lua [hunk 2]
```diff
@@ -390,13 +399,16 @@ else
                 rotation = -20,
                 debug = config.debugPoly,
                 onEnter = function()
+                    if QBX.PlayerData.job.type ~= 'ems' then return end
                     local label = QBX.PlayerData.job.onduty and locale('text.onduty_button') or locale('text.offduty_button')
                     lib.showTextUI(label)
                 end,
                 onExit = function()
+                    if QBX.PlayerData.job.type ~= 'ems' then return end
```
Comment (solareon) [inline client/job.lua]:
```suggestion
                    local _, text = lib.isTextUIOpen()
                    if text == QBX.PlayerData.job.onduty and locale('text.onduty_button') or locale('text.offduty_button')
```
Better solu

File: client/job.lua [hunk 3]
```diff
@@ -409,14 +421,17 @@ else
                 rotation = -20,
                 debug = config.debugPoly,
                 onEnter = function()
+                    if QBX.PlayerData.job.type ~= 'ems' then return end
                     if QBX.PlayerData.job.onduty then
                         lib.showTextUI(locale('text.pstash_button'))
                     end
                 end,
                 onExit = function()
+                    if QBX.PlayerData.job.type ~= 'ems' then return end
```
Comment (solareon) [inline client/job.lua]:
```suggestion
```
Don't let people get the textui stuck if they lose the EMS job while in the zone

File: client/job.lua [hunk 4]
```diff
@@ -432,14 +447,17 @@ else
                     rotation = -20,
                     debug = config.debugPoly,
                     onEnter = function()
+                        if QBX.PlayerData.job.type ~= 'ems' then return end
                         if QBX.PlayerData.job.onduty then
                             lib.showTextUI(locale('text.armory_button'))
                         end
                     end,
                     onExit = function()
+                        if QBX.PlayerData.job.type ~= 'ems' then return end
```
Comment (solareon) [inline client/job.lua]:
```suggestion
```
Don't let people get the textui stuck if they lose the EMS job while in the zone

**Review by solareon — CHANGES_REQUESTED** [review]

**Pull request merged**